python/src/scipp/compat/mantid.py

```diff
@@ -454,14 +454,12 @@ def from_mantid(workspace, **kwargs):
 
         monitors = convert_monitors_ws(monitor_ws, converter, **kwargs)
         for name, monitor in monitors:
+            # Remove redundant information that is duplicated from workspace
+            del monitor.labels['sample_position']
+            del monitor.labels['detector_info']
+            del monitor.attrs['run']
+            del monitor.attrs['sample']
             dataset.attrs[name] = sc.Variable(value=monitor)
-
-        # Remove some redundant information that is duplicated from workspace
-        mon = dataset.attrs["monitors"].value
-        del mon.labels['sample_position']
-        del mon.labels['detector_info']
-        del mon.attrs['run']
-        del mon.attrs['sample']
     except RuntimeError:
         pass
```

Comment:
@OwenArnold This silent `pass` was probably the reason why this bug was overlooked. I tried to understand why this whole block is even wrapped in `try`, do you understand the reason? Maybe it can be avoided entirely.

Reply:
I'll investigate that try block